components/basecamp/basecamp.app.mjs

```diff
@@ -1,11 +1,342 @@
+      options: constants.RECORDING_TYPE_OPTS,
+    },
+    peopleIds: {
+      type: "string[]",
+      label: "People",
+      description: "An array of all people visible to the current user.",
+      async options({
+        accountId,
+        projectId,
+      }) {
+        const people = await this.getPeople({
+          accountId,
+          projectId,
+        });
+        return people.
+          map(({
+            id,
+            name,
+          }) => ({
+            label: name,
+            value: id,
+          }));
+      },
+    },
+    todoSetId: {
+      type: "string",
+      label: "Todo Set Id",
+      description: "The ID of the todo set.",
+      async options({
+        accountId,
+        projectId,
+      }) {
+        const project = await this.getProject({
+          accountId,
+          projectId,
+        });
+        return project.dock.filter((dock) => dock.name === "todoset" && dock.enabled === true)
+          .map(({
+            id,
+            title,
+          }) => ({
+            label: title,
+            value: id,
+          }));
+      },
+    },
+    campfireId: {
+      type: "string",
+      label: "Campfire Id",
+      description: "The ID of the campfire.",
+      async options({
+        accountId,
+        projectId,
+      }) {
+        const project = await this.getProject({
+          accountId,
+          projectId,
+        });
+        return project.dock.filter((dock) => dock.name === "chat" && dock.enabled === true)
+          .map(({
+            id,
+            title,
+          }) => ({
+            label: title,
+            value: id,
+          }));
+      },
+    },
+    todoListId: {
+      type: "string",
+      label: "Todo List Id",
+      description: "The ID of the todo list.",
+      async options({
+        accountId,
+        projectId,
+        todoSetId,
+      }) {
+        const todoslists = await this.getTodoLists({
+          accountId,
+          projectId,
+          todoSetId,
+        });
+        return todoslists
+          .map(({
+            id,
+            title,
+          }) => ({
+            label: title,
+            value: id,
+          }));
+      },
+    },
+    messageTypeId: {
+      type: "string",
+      label: "Message Types",
+      description: "Select a type for the message.",
+      async options({
+        accountId,
+        projectId,
+      }) {
+        const types = await this.getMessageTypes({
+          accountId,
+          projectId,
+        });
+        return types.map(({
+          id,
+          name,
+        }) => ({
+          label: name,
+          value: id,
+        }));
+      },
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async getAccounts({ $ = this }) {
+      const { accounts } = await axios($, {
+        url: "https://launchpad.37signals.com/authorization.json",
```

Comment:
Is the url correct? Just asking to make sure

Reply:
yes, this url returns the accounts for the oAuth session, so we can use the accountId to the rest of the requests...

you can see more [here](https://launchpad.37signals.com/authorization.json)